scanpy/plotting/anndata.py

```diff
@@ -718,3 +762,120 @@ def clustermap(
+    var_names : `str` or list of `str`
+        var_names should be a valid subset of  `.var_names`.
+    groupby : `str` or `None`, optional (default: `None`)
+        The key of the observation grouping to consider. It is expected that groupby is
+        a categorical. If groupby is not a categorical observation, it would be
+        subdivided into `num_categories`.
+    log : `bool`, optional (default: `False`)
+        Use the log of the values
+    use_raw : `bool`, optional (default: `True`)
+        Use `raw` attribute of `adata` if present.
+    num_categories : `int`, optional (default: `7`)
+        Only used if groupby observation is not categorical. This value determines
+        the number of groups into which the groupby observation should be subdivided.
+    show : bool, optional (default: `None`)
+         Show the plot.
+    save : `bool` or `str`, optional (default: `None`)
+        If `True` or a `str`, save the figure. A string is appended to the
+        default filename. Infer the filetype if ending on {{'.pdf', '.png', '.svg'}}.
+    **kwargs : keyword arguments
+        Are passed to `seaborn.heatmap`.
+
+    Returns
+    -------
+    A list of `matplotlib.Axes` where the first ax is the groupby categories colorcode, the
+    second axis is the heatmap and the third axis is the colorbar.
+
+    """
+    from scipy.sparse import issparse
+    sanitize_anndata(adata)
+    if isinstance(var_names, str):
+        var_names = [var_names]
+    if groupby is not None:
+        if groupby not in adata.obs_keys():
+            raise ValueError('groupby has to be a valid observation. Given value: {}, '
+                             'valid observations: {}'.format(groupby, adata.obs_keys()))
+
+    if use_raw:
+        matrix = adata.raw[:, var_names].X
+    else:
+        matrix = adata[:, var_names].X
+
+    if issparse(matrix):
+        matrix = matrix.toarray()
+    if log:
+        matrix = np.log1p(matrix)
+
+    obs_tidy = pd.DataFrame(matrix, columns=var_names)
+    if groupby is None:
+        x = 'variable'
+    else:
+        if not is_categorical_dtype(adata.obs[groupby]):
+            # if the groupby column is not categorical, turn it into one
+            # by subdividing into 7 categories
+            categorical = pd.cut(adata.obs[groupby], num_categories)
+        else:
+            categorical = adata.obs[groupby]
+        obs_tidy.set_index(categorical, groupby, inplace=True)
+        x = obs_tidy.index.categories
+
+    height = 12
+    width = len(var_names) * 0.3 + 2
+    ax_frac2width = 0.010 * width
+    fig, axs = pl.subplots(nrows=1, ncols=3, sharey=False,
+                           figsize=(width, height), gridspec_kw={'width_ratios': [ax_frac2width, width, ax_frac2width]})
+    groupby_ax = axs[0]
+    heatmap_ax = axs[1]
+    heatmap_cbar_ax = axs[2]
+
+    if groupby:
+        obs_tidy = obs_tidy.sort_index()
+
+    groupby_ax.imshow(np.matrix(obs_tidy.index.codes).T, aspect='auto')
```

Comment:
Hi @fidelram! You added this `np.matrix` call.

Do you know if there’s any reason you didn’t use an ndarray? This line is throwing a warning now.